### PR TITLE
Weaker infra failure conditions, notify on build failures

### PR DESCRIPTION
**Pull request by dhdavvie**
If we have build failures, we should notify the bug. We should also not flag the try push for manual fix if we have some builds and subsequent tests that completed. 

Should also add build failure notification in `try_notify` but will probably do that in another PR.

### CODE REVIEW

File: sync/trypush.py [hunk 1]
```diff
@@ -350,19 +351,44 @@ def stability(self):
...
+        msg += "\n".join(task.get("task", {}).get("metadata", {}).get("name")
+                         for task in failed.tasks)
+        env.bz.comment(self.bug, msg)
+
     @infra_fail.setter
     @mut()
     def infra_fail(self, value):
-        """Is the current try push a stability test"""
+        """Set the status of this push's infrastructure failure state"""
         self["infra-fail"] = value
```
Comment (jgraham) [inline sync/trypush.py]: Can we return if the value isn't changing?

File: sync/trypush.py [hunk 2]
```diff
@@ -554,6 +585,14 @@ def has_failures(self):
             return any(task.get("status", {}).get("state") == tc.FAIL for task in wpt_tasks)
         return False
 
+    def has_completed_tests(self):
+        """Check if all the wpt tasks in a try push completed"""
+        wpt_tasks = self.wpt_tasks.filter(tc.is_test)
+        if wpt_tasks:                                          # tc.EXCEPTION ??
```
Comment (jgraham) [inline sync/trypush.py]: Nah, if they all end up in Exception that's pretty bad. So this is fine.

File: sync/trypush.py (hunk 1 above)
Comment (dhdavvie) [reply sync/trypush.py]: Would be a good idea, will implement that.

File: sync/downstream.py [hunk 3]
```diff
@@ -176,7 +176,13 @@ def next_action(self):
             # If we have infra failure, flag for human intervention. Retrying stability
             # runs would be very costly
             if latest_try_push.infra_fail:
-                return DownstreamAction.manual_fix
+                tasks = latest_try_push.tasks()
+
+                # Check if we had any successful tests
+                if tasks.has_completed_tests():
```
Comment (jgraham) [inline sync/downstream.py]: I think this should use `tasks.wpt_tasks` to allow for the possibility of `source-test` jobs being considered tests.

Comment (dhdavvie) [reply sync/downstream.py]: The `tasks.has_completed_tests` method filters for wpt tests like so `wpt_tasks = self.wpt_tasks.filter(tc.is_test)`. Will this filter out `source-test` jobs?

Comment (jgraham) [reply sync/downstream.py]: Oh it seems like they're kind `source-test` so I guess this is fine.

**Pull request merged**